### PR TITLE
fix the leaderboard metric in onmt_helpers

### PR DESCRIPTION
**Pull request by hao-fang**
This PR does not fix the leaderboard.
It only fixes the way I use `try_round_trip` in `onmt_helpers.create_onmt_prediction_report`. See inline comments.

### CODE REVIEW

File: src/dataflow/core/linearize.py [hunk 1]
```diff
@@ -18,6 +18,17 @@
 from dataflow.core.sexp import Sexp
 
 
+def to_canonical_form(tokenized_lispress: str) -> str:
```
Comment (hao-fang) [inline src/dataflow/core/linearize.py:21]: added this to handle tokenized lispress

File: src/dataflow/onmt_helpers/create_onmt_prediction_report.py [hunk 2]
```diff
@@ -45,17 +45,22 @@
...
+    # The tokenized gold lispress.
     gold: str
+    # The tokenized predicted lispress.
     prediction: str
     program_execution_oracle: ProgramExecutionOracle
 
     @property
     def gold_canonical(self) -> str:
-        return try_round_trip(self.gold)
+        return to_canonical_form(self.gold)
```
Comment (hao-fang) [inline src/dataflow/onmt_helpers/create_onmt_prediction_report.py:56]: `try_round_trip` expects untokenized lispress str, but `gold` and `prediction` here are tokenized.
Also, this should not throw exceptions for `gold`, but for `prediction` we return `_PARSE_ERROR_LISPRESS` if it fails to parse.

**Review by hao-fang — COMMENTED** [review]
I checked the final eval results on dev and they are still the same.

File: src/dataflow/core/linearize.py [hunk 3]
```diff
@@ -12,12 +12,24 @@
...
 from dataflow.core.program_utils import Idx, OpType
 from dataflow.core.sexp import Sexp
 
 
+def to_canonical_form(tokenized_lispress: str) -> str:
+    """Returns canonical form of a tokenized lispress.
+
+    The canonical form is un-tokenized and compact; it also sorts named arguments in alphabetical order.
+    """
+    lispress = seq_to_lispress(tokenized_lispress.split(" "))
```
Comment (sammthomson) [inline src/dataflow/core/linearize.py:27]: this is where the leading and trailing spaces are getting trimmed, right?

Comment (hao-fang) [reply src/dataflow/core/linearize.py:27]: Yes, I think so.
So you will need to change the leaderboard code accordingly.

File: src/dataflow/core/linearize.py [hunk 4]
```diff
@@ -12,12 +12,24 @@
...
+    program, _ = lispress_to_program(lispress, 0)
+    round_tripped = program_to_lispress(program)
+    return render_compact(round_tripped)
```
Comment (sammthomson) [inline src/dataflow/core/linearize.py:30]: is this the same as?
```suggestion
    deformatted = sexp_deformatter(tokenized_lispress.split(" "))
    return try_round_trip(deformatted)
```

**Review by sammthomson — APPROVED** [review]
thanks!

File: src/dataflow/core/linearize.py (hunk 4 above)
Comment (hao-fang) [reply src/dataflow/core/linearize.py:30]: Hmm... the type doesn't match?
`sexp_deformatter` returns `List[str]`, but `try_round_trip` takes `str`.
We may change change `try_round_trip` to take `Lispress` by moving `parse_lispress(lispress_str)` outside the `try_round_trip`, then we can reduce some lines of the code here.

Comment (sammthomson) [reply src/dataflow/core/linearize.py:30]: oh yeah, sorry, wrote without testing. no, i want the parse_lispress inside the try. i think it's fine to leave it how you have it

**Pull request merged**